### PR TITLE
libtorrent: fix cmake imported target + bump dependencies + check boost with system

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **libtorrent/all**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
There are several issues here:

- Windows: long path error, but it shouldn't happen:

```
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(181,5): error MSB4018: The "GetOutOfDateItems" task failed unexpectedly. [C:\J\w\cci_PR-2684@2\3\c161b7dc-8972-4101-ae4b-8932c26fe86e\.conan\data\libtorrent\1.2.3\_\_\build\bc22d030bf4291944fda5f92aabb480c8266563c\build_subfolder\source_subfolder\torrent-rasterbar.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(181,5): error MSB4018: System.InvalidOperationException: The item metadata "%(FullPath)" cannot be applied to the path "torrent-rasterbar.dir\Release\torrent-.D52EB9B3.tlog\ParallelCustomBuild.command.1.tlog". Path: torrent-rasterbar.dir\Release\torrent-.D52EB9B3.tlog\ParallelCustomBuild.command.1.tlog exceeds the OS max path limit. The fully qualified file name must be less than 260 characters. [C:\J\w\cci_PR-2684@2\3\c161b7dc-8972-4101-ae4b-8932c26fe86e\.conan\data\libtorrent\1.2.3\_\_\build\bc22d030bf4291944fda5f92aabb480c8266563c\build_subfolder\source_subfolder\torrent-rasterbar.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(181,5): error MSB4018:    at Microsoft.Build.Shared.ErrorUtilities.ThrowInvalidOperation(String resourceName, Object[] args) [C:\J\w\cci_PR-2684@2\3\c161b7dc-8972-4101-ae4b-8932c26fe86e\.conan\data\libtorrent\1.2.3\_\_\build\bc22d030bf4291944fda5f92aabb480c8266563c\build_subfolder\source_subfolder\torrent-rasterbar.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(181,5): error MSB4018:    at Microsoft.Build.Shared.FileUtilities.ItemSpecModifiers.GetItemSpecModifier(String currentDirectory, String itemSpec, String definingProjectEscaped, String modifier, String& fullPath) [C:\J\w\cci_PR-2684@2\3\c161b7dc-8972-4101-ae4b-8932c26fe86e\.conan\data\libtorrent\1.2.3\_\_\build\bc22d030bf4291944fda5f92aabb480c8266563c\build_subfolder\source_subfolder\torrent-rasterbar.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(181,5): error MSB4018:    at Microsoft.Build.Utilities.TaskItem.Microsoft.Build.Framework.ITaskItem2.GetMetadataValueEscaped(String metadataName) [C:\J\w\cci_PR-2684@2\3\c161b7dc-8972-4101-ae4b-8932c26fe86e\.conan\data\libtorrent\1.2.3\_\_\build\bc22d030bf4291944fda5f92aabb480c8266563c\build_subfolder\source_subfolder\torrent-rasterbar.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(181,5): error MSB4018:    at Microsoft.Build.Utilities.TaskItem.GetMetadata(String metadataName) [C:\J\w\cci_PR-2684@2\3\c161b7dc-8972-4101-ae4b-8932c26fe86e\.conan\data\libtorrent\1.2.3\_\_\build\bc22d030bf4291944fda5f92aabb480c8266563c\build_subfolder\source_subfolder\torrent-rasterbar.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(181,5): error MSB4018:    at Microsoft.Build.CPPTasks.GetOutOfDateItems.Execute() [C:\J\w\cci_PR-2684@2\3\c161b7dc-8972-4101-ae4b-8932c26fe86e\.conan\data\libtorrent\1.2.3\_\_\build\bc22d030bf4291944fda5f92aabb480c8266563c\build_subfolder\source_subfolder\torrent-rasterbar.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(181,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [C:\J\w\cci_PR-2684@2\3\c161b7dc-8972-4101-ae4b-8932c26fe86e\.conan\data\libtorrent\1.2.3\_\_\build\bc22d030bf4291944fda5f92aabb480c8266563c\build_subfolder\source_subfolder\torrent-rasterbar.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(181,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext() [C:\J\w\cci_PR-2684@2\3\c161b7dc-8972-4101-ae4b-8932c26fe86e\.conan\data\libtorrent\1.2.3\_\_\build\bc22d030bf4291944fda5f92aabb480c8266563c\build_subfolder\source_subfolder\torrent-rasterbar.vcxproj]
```

What is this `c161b7dc-8972-4101-ae4b-8932c26fe86e` folder?

- Macos: test_package fail because of https://github.com/conan-io/conan/issues/7549 I think. I can try to use `cmake_find_package` generator instead of `cmake_find_package_multi` in test_package

- Debug builds (all os): test_package fails due do assertions. @madebr Any idea?

**Comment by madebr** [comment]
>     * Debug builds (all os): test_package fails due do assertions. @madebr Any idea?

It's a problem with boost asio, I think.
- boost 1.72.0: OK
- boost 1.73.0: OK
- boost 1.74.0: NOK

Or our test package is wrong.
I would follow the assertion's advice and open a github issue.

**Comment by SpaceIm** [comment]
@madebr @prince-chrismc @ericLemanissier @SSE4 @uilianries @danimtb This PR is ready for review

File: recipes/libtorrent/all/conanfile.py [hunk 1]
```diff
@@ -88,13 +89,14 @@ def _configure_cmake(self):
...
-        self._cmake.configure(build_folder=self._build_subfolder)
+        self._cmake.configure()
         return self._cmake
 
     def _patch_sources(self):
-        for patch_data in self.conan_data["patches"][self.version]:
+        for patch_data in self.conan_data.get("patches", {}).get(self.version, []):
             tools.patch(**patch_data)
 
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"), "/W4", "")
```
Comment (prince-chrismc) [inline recipes/libtorrent/all/conanfile.py:99]: Why do we need to disable warnings?

File: recipes/libtorrent/all/test_package/CMakeLists.txt [hunk 2]
```diff
@@ -8,6 +8,8 @@ if(WIN32)
     add_definitions(-D_WIN32_WINNT=0x0601)
```
Comment (prince-chrismc) [inline recipes/libtorrent/all/test_package/CMakeLists.txt:8]: (optional) This should be an include `SDKDDKVer.h` in the cpp

**Review by prince-chrismc — APPROVED** [review]

File: recipes/libtorrent/all/conanfile.py (hunk 1 above)
Comment (madebr) [reply recipes/libtorrent/all/conanfile.py:99]: I think it is done for convenience (`/W4` spews **a lot** of warnings).

Comment (SpaceIm) [reply recipes/libtorrent/all/conanfile.py:99]: that's it

**Review by SSE4 — APPROVED** [review]

**Review by danimtb — APPROVED** [review]

**Pull request merged**